### PR TITLE
Fuse attentions for some models

### PR DESCRIPTION
**Pull request by wangyems**
fuse attention for:
1. A bert model that converted using tf2onnx

### CODE REVIEW

**Comment by tianleiwu** [comment]
Please add some test cases to test_optimizer.py

File: onnxruntime/python/tools/transformers/onnx_model_bert_tf.py [hunk 1]
```diff
@@ -315,4 +421,5 @@ def remove_reshape_before_first_attention(self):
 
     def postprocess(self):
         self.remove_reshape_before_first_attention()
-        self.prune_graph()
+        # Temporary work around for the following comment as it will cause topological issues for a bert model
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_model_bert_tf.py:424]: Do you know why prone graph cause issues? It used to be fine.

File: onnxruntime/python/tools/transformers/onnx_model_bert_tf.py [hunk 2]
```diff
@@ -295,6 +295,112 @@ def process_embedding(self):
...
+
+            is_same_root = self.check_attention_input(matmul_q, matmul_k, matmul_v, parent, output_name_to_node)
+            if is_same_root:
+                mask_index = self.attention_mask.process_mask(mask_nodes[-1].input[0])
+                logger.debug("Create an Attention node.")
+                attention_node = self.attention_fusion.create_attention_node(mask_index, matmul_q, matmul_k, matmul_v,
+                                                                             add_q, add_k, add_v, parent.output[0],
+                                                                             reshape_qkv.output[0])
+                if parent.op_type == 'Reshape':
+                    # Temporary work around
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/onnx_model_bert_tf.py:378]: Add more information to the comment, like what issue and why this work around is needed.

Comment (wangyems) [reply onnxruntime/python/tools/transformers/onnx_model_bert_tf.py:378]: Sure

**Comment by wangyems** [comment]
move to a new PR for easy rebasing

**Pull request closed without merging**